WalletWasabi.Fluent/ViewModels/CoinJoinProfiles/CoinJoinProfilesViewModel.cs

```diff
@@ -75,22 +75,26 @@ private async Task OnManualSetupAsync()
 		}
 	}
 
-	private void OnNext(KeyManager keyManager, bool isNewWallet)
+	private void OnNext(IWalletModel walletModel, bool isNewWallet)
 	{
 		var selected = SelectedProfile ?? SelectedManualProfile ?? Profiles.First();
 
-		keyManager.RedCoinIsolation = selected.RedCoinIsolation;
-		keyManager.SetAnonScoreTarget(selected.AnonScoreTarget, toFile: false);
-		keyManager.SetFeeRateMedianTimeFrame(selected.FeeRateMedianTimeFrameHours, toFile: false);
-		keyManager.IsCoinjoinProfileSelected = true;
+		using (walletModel.Settings.BatchChanges())
+		{
+			walletModel.Settings.RedCoinIsolation = selected.RedCoinIsolation;
+			walletModel.Settings.AnonScoreTarget = selected.AnonScoreTarget;
+			walletModel.Settings.FeeRateMedianTimeFrameHours = selected.FeeRateMedianTimeFrameHours;
+			walletModel.Settings.IsCoinjoinProfileSelected = true;
+		}
```

Comment:
I am not 100% sure this is ok. Using a disposable for this.

What do you think about a solution like:
```c#
public void BatchChanges(Action<IWalletSettingsModel> action)
{
	_batchChanges = true;
	action.Invoke(this);
	_batchChanges = false;
	Save();
}
```
So it can be used like:
```c#
walletModel.Settings.BatchChanges(x =>
{
	x.RedCoinIsolation = selected.RedCoinIsolation;
	x.AnonScoreTarget = selected.AnonScoreTarget;
	x.FeeRateMedianTimeFrameHours = selected.FeeRateMedianTimeFrameHours;
	x.IsCoinjoinProfileSelected = true;
});
```

So it make it sure you can only make changes on the instance that you called `BatchChanges` on.

Reply:
@soosr this API was removed on #10730. I figured out it doesn't really make sense because the only place where *immediate save* is needed for wallet settings is `CoinJoinSettingsViewModel`, and that only contains 2 editable properties (`PlebStopThreshold` and `AutoCoinjoin` flag) so I decided to manually invoke `Save()` upon change of these properties in that ViewModel, and get rid of the autosave feature in the model altogether. This reduces complexity significantly.